### ISSUE TITLE
Question on result

### ISSUE THREAD

**Issue by Masa-Shin** [emilwallner/Screenshot-to-code-in-Keras#20]
The program worked without errors, but the result was far from I expected ( maybe I did something wrong? ). 

here is the image I used:
![68747470733a2f2f71696974612d696d6167652d73746f72652e73332e616d617a6f6e6177732e636f6d2f302f3132333830392f64396337356562652d626238622d666638652d373663612d6365396463343330386262302e6a706567](https://user-images.githubusercontent.com/34234442/35431692-d36ec0b4-02c0-11e8-826b-aac8168ef7c2.jpg)

And here is the result: https://jsfiddle.net/b1zt7vsh/

What I did:
1. Put the image under Screenshot-to-code-in-Keras/floydhub/HTML/resources/images folder.
2. Rewrote the line 131 of HTML.ipynb (change the path to the image).
3. Rewrote the line 190 of HTML.ipynb  (epochs to be 300).
4. Ran All cells of HTML.ipynb.

The value of loss function was below 0.001.

I would greatly appreciated it if you could tell me what I did was correct or not.

**Comment by emilwallner** [comment]
@Masa-Shin  Thanks for your question. 

As mentioned in the article, the HTML version does not generalize on new images. The Bootstrap version generalizes on new images but with a capped vocabulary. The evaluation images for the bootstrap version are under /data/eval/ . You can test it here: floydhub/Bootstrap/test_model_accuracy.ipynb

If you want to train it to generalize on a more advanced vocabulary, I'd recommend customizing it to work on the HTML set provided here: https://github.com/harvardnlp/im2markup (on floydhub: --data emilwallner/datasets/100k-html:data) 

After that, I'd recommend creating a new dataset. Create a script that generates random websites, say starting with newsletters or blog layouts. Then you can add optical character recognition, fonts, colors and div sizes as you go. 

If you build a version for the harvardnlp dataset or a script that generates websites, please make a pull request. 

Let me know if there is anything else I can help with.

**Comment by Masa-Shin** [comment]
I understood fully. Thank you so much for the detailed explanations (and the wonderful product)!

**Comment by Masa-Shin** [comment]
I understood. I will try the Bootstrap version. Thank you very much for detailed explanations (and the wonderful product)!
